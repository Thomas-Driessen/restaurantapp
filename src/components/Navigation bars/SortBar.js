import React from "react";
import AppBar from "@material-ui/core/AppBar";
import { Button, makeStyles } from "@material-ui/core";
import ButtonGroup from "@material-ui/core/ButtonGroup";
import Container from "@material-ui/core/Container";

const useStyles = makeStyles(
  (theme) => ({
    buttonPressed: {
      background: theme.palette.primary.main,
      borderRadius: 3,
      border: 0,
      color: "white",
      height: 48,
      padding: "0 30px",
    },
  }),
  { withTheme: true }
);

const SortBar = (props) => {
  const classes = useStyles();
      return (
    <div>
      <AppBar position="static" color="transparent">
        <Container disableGutters>
          <ButtonGroup
            fullWidth
            variant="text"
            size="large"
            color="primary"
            aria-label="text primary button group"
          >
            <Button
              onClick={props.showFoods}
              className={
                props.productType === "Food" ? classes.buttonPressed : ""
              }
            >
              Food
            </Button>
            <Button
              onClick={props.showDrinks}
              className={
                props.productType === "Drink" ? classes.buttonPressed : ""
              }
            >
              Drinks
            </Button>
          </ButtonGroup>
        </Container>
      </AppBar>
    </div>
  );
};

export default SortBar;
