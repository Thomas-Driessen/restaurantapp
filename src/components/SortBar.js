import React from 'react';
import AppBar from '@material-ui/core/AppBar'
import ToolBar from '@material-ui/core/Toolbar'
import { Button } from '@material-ui/core';

const SortBar = (props) => {
    return(
        <div>
            <AppBar position="static" color="contrastText">
                <ToolBar>
                    <Button  onClick={props.showFoods} variant="contained" color="default">
                    Food
                    </Button>
                    <Button onClick={props.showDrinks} variant="contained" color="default">
                    Drinks
                    </Button>
                    <Button  variant="contained" color="default">
                    Most Liked
                    </Button>
                </ToolBar>
            </AppBar>
        </div>
    )
}

export default SortBar;