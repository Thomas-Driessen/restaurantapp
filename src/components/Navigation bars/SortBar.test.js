import React from 'react';
import { render } from '@testing-library/react';
import SortBar from './SortBar';
import renderer from 'react-test-renderer';

it('SortBar renders correctly', () => {
  const tree = renderer
    .create(<SortBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Render SortBar Food Button', () => {
  const { getByText } = render(<SortBar />);
  const linkElement = getByText(/Food/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render SortBar Drink Button', () => {
  const { getByText } = render(<SortBar />);
  const linkElement = getByText(/Drink/i);
  expect(linkElement).toBeInTheDocument();
});

test('Render SortBar Most Liked Button', () => {
  const { getByText } = render(<SortBar />);
  const linkElement = getByText(/Most Liked/i);
  expect(linkElement).toBeInTheDocument();
});

afterAll(() => {
  global.gc && global.gc()
})