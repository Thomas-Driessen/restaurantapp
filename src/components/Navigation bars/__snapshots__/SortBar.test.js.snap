// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortBar renders correctly 1`] = `
<div>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorTransparent MuiPaper-elevation4"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Food
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Drinks
        </span>
      </button>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Most Liked
        </span>
      </button>
    </div>
  </header>
</div>
`;
